multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/SupportedParametersChecker.java

```diff
@@ -0,0 +1,114 @@
+package org.cloudfoundry.multiapps.controller.process.util;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.cloudfoundry.multiapps.controller.core.model.SupportedParameters;
+import org.cloudfoundry.multiapps.mta.model.DeploymentDescriptor;
+import org.cloudfoundry.multiapps.mta.model.Hook;
+import org.cloudfoundry.multiapps.mta.model.Module;
+import org.cloudfoundry.multiapps.mta.model.RequiredDependency;
+import org.cloudfoundry.multiapps.mta.model.Resource;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+
+@Named("supportedParametersChecker")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class SupportedParametersChecker {
```

Comment:
Could this whole recursive parameter checking be done via the Visitor pattern we already use for the Validators? (not saying it's a good idea to do it, but it might be better?)